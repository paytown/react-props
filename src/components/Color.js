import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, red, green, blue }) {
  const style = {
    backgroundColor: `rgba(${red}, ${green}, ${blue}, 1)`,
    borderRadius: '10px',
    width: '50px',
    height: '50px'
  };

  return (
    <main>
      {name && (<>
        <h1>Name</h1>
        <h2>{name}</h2>
      </>)}

      <div style={style}></div>

      <h3>RGB</h3>
      <ul>
        <li>Red: {red}</li>
        <li>Green: {green}</li>
        <li>Blue: {blue}</li>
      </ul>
    </main>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  red: PropTypes.number.isRequired,
  green: PropTypes.number.isRequired,
  blue: PropTypes.number.isRequired,
};

export default Color;
