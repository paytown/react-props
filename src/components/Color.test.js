import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('colors', () => {
  it('renders color dynamically', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
