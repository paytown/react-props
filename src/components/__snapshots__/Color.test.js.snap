// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors renders color dynamically 1`] = `
<main>
  <div
    style={
      Object {
        "backgroundColor": "rgba(undefined, undefined, undefined, 1)",
        "borderRadius": "10px",
        "height": "50px",
        "width": "50px",
      }
    }
  />
  <h3>
    RGB
  </h3>
  <ul>
    <li>
      Red: 
    </li>
    <li>
      Green: 
    </li>
    <li>
      Blue: 
    </li>
  </ul>
</main>
`;
